### DISCUSSION TITLE
The target encoder is always a copy of the online encoder?

### DISCUSSION THREAD

**Discussion by lishen** [lucidrains/byol-pytorch#80] [General]
https://github.com/lucidrains/byol-pytorch/blob/6717204748c2a4f4f44b991d4c59ce5b99995582/byol_pytorch/byol_pytorch.py#L258
https://github.com/lucidrains/byol-pytorch/blob/6717204748c2a4f4f44b991d4c59ce5b99995582/byol_pytorch/byol_pytorch.py#L225
These two lines of code implicate that the target encoder is always set to be a copy of the online encoder no matter the momentum is turned on or off. Am I reading this correctly?

**Comment by lb-97** [comment]
Shouldn't it rather be a shallow copy? Otherwise the weights of encoder are copied at every forward irrespective of momentum being on
